webapp/channels/src/components/suggestion/suggestion.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useCallback} from 'react';
+
+export interface SuggestionProps<Item> {
+    // eslint-disable-next-line react/no-unused-prop-types
+    item: Item;
+
+    term: string;
+    matchedPretext: string;
+    isSelection: boolean;
+
+    children?: React.ReactNode;
+    onClick: (term: string, matchedPretext: string) => void;
+    onMouseMove: (term: string) => void;
+
+    'data-testid'?: string;
+    id?: string;
+    role?: string;
+    tabIndex?: number;
```

Comment:
We can't just have these props extend `React.HTMLAttributes` because we overload `onClick` and `onMouseMove` with different signatures

Reply:
Wouldn't the `Omit` typescript utility type work for that? Something like:
`Omit<React.HTMLAttributes, 'onClick' | 'onMouseMove'> & {onClick: ..., onMouseMove: ...}`

Reply:
Good idea! I don't know how I missed that when I used `Omit` elsewhere in this PR